subsys/console/getchar.c

```diff
@@ -117,3 +120,25 @@ void console_init(void)
 	uart_irq_callback_set(uart_dev, uart_isr);
 	uart_irq_rx_enable(uart_dev);
 }
+
+#elif defined(CONFIG_SOC_POSIX)
```

Comment:
To base this on the SOC config is a bad idea.
The idea was that only the "board" defines peripherals. Different ARCH_POSIX based boards use different means to route out their console input/output, only native_posix uses directly stdin and stdout.

That's why the native_posix driver has it's own configuration options.

Reply:
Please suggest a better (but existing) define.

> That's why the native_posix driver has it's own configuration options.

Yes, that's good, but seems to be getting too fine-grained. I'd follow YAGNI approach and not create too many config options until there're actually more POSIX "boards". (I may be missing something here.)

Reply:
Wouldn't the obvious one which meets your requirements be BOARD_NATIVE_POSIX?
But the reason why there is a decoupling between the board and the drivers it selects is so other boards can share some of those drivers.

Reply:
Probably yes, but I don't think we have many #ifdef BOARD_xxx in generic code, and such a code does look strange. I believe that was the reason I put SOC_xxx and consulting with you on the best approach, thanks.